qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -80,6 +80,7 @@ def run(self, dag):
             new_dag.add_qreg(qreg)
         for creg in dag.cregs.values():
             new_dag.add_creg(creg)
+        new_dag._global_phase = dag._global_phase
```

Comment:
Does every pass that creates a new `DAGCircuit` (rather than modifying the input `dag`) need to be updated this way? (This might be a good opportunity to consolidate the "create a new empty dag over the registers of this other dag" pattern that shows up occasionally, e.g. https://github.com/Qiskit/qiskit-terra/blob/f58e747292d7740c3697751d7ba2bb88af0d8b06/qiskit/transpiler/passes/routing/lookahead_swap.py#L300 ).

Reply:
I'll try one on `DAGCircuit`.